CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/io/ParameterIOUtilitiesNGTest.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2010-2021 Australian Signals Directorate
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.views.dataaccess.io;
+
+import au.gov.asd.tac.constellation.graph.Graph;
+import au.gov.asd.tac.constellation.graph.WritableGraph;
+import au.gov.asd.tac.constellation.plugins.parameters.PluginParameter;
+import au.gov.asd.tac.constellation.plugins.parameters.PluginParameters;
+import au.gov.asd.tac.constellation.views.dataaccess.DataAccessState;
+import au.gov.asd.tac.constellation.views.dataaccess.panes.GlobalParametersPane;
+import au.gov.asd.tac.constellation.views.dataaccess.panes.QueryPhasePane;
+import java.util.Map;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.embed.swing.JFXPanel;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.Tab;
+import javafx.scene.control.TabPane;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * @author mimosa2
+ */
+public class ParameterIOUtilitiesNGTest {
+
+    public ParameterIOUtilitiesNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() {
+    }
+
+    @AfterClass
+    public static void tearDownClass() {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of saveDataAccessState method, of class ParameterIOUtilities.
+     */
+    @Test
+    public void testsaveDataAccessState() throws Exception {
+        System.out.println("testsaveDataAccessState");
+        new JFXPanel();
```

Comment:
I assume this is to instantiate the JFX toolkit? If so, are you able to use the testfx functions to set up the JFX environment? This will help in the future if more functions are tested from this class

Reply:
I tried to use JFX toolkit then I got an error saying couldn't mock(TabPane.class). Let me know is there a workaround.?

Reply:
@mimosa2 check out some of the changes made in #1282, namely the code in `setUpClass()` and `tearDownClass()`. That will be what to do instead of `new JFXPanel()`

Reply:
That is exactly what I did earlier. Then I got the above error. I will check again

Reply:
Worked after moving the code to setUpMethod() and tearDownMetod(). Thanks @antares1470.